src/components/PasteButton.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import { GestureResponderEvent, StyleSheet, Text } from 'react-native'
+import Touchable from 'src/components/Touchable'
+import { Colors } from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { useClipboard } from 'src/utils/useClipboard'
+
+interface Props {
+  value: string
+  setValue: React.Dispatch<React.SetStateAction<string>>
+  onPress?: (event: GestureResponderEvent) => void
+}
+
+export const PasteButton = ({ value, setValue, onPress }: Props) => {
+  const { t } = useTranslation()
+  const [, , getFreshClipboardContent] = useClipboard()
+
+  const pasteClipboard = async (event: GestureResponderEvent) => {
+    const content = await getFreshClipboardContent()
+    setValue(content)
+    onPress?.(event)
+  }
+
+  // when a value is already set, remove the button
+  if (value) return null
+
+  return (
+    <Touchable onPress={pasteClipboard}>
```

Comment:
```suggestion
    <Touchable onPress={handlePasteClipboard}>
```
or `onPasteClipboard`. we usually prefix any handlers with `on` or `handle`. `on` is usually used for passed through props, and `handle` for any handlers that the component declares by itself (although we've not been so consistent about the latter, sometimes the `on` prefix is used)